guide/src/reference/attributes/on-rust-exports/main.md

```diff
@@ -0,0 +1,33 @@
+# `main`
+
+When attached to the `main` function this attribute will adjust it to properly
+throw errors if they can be.
+
+```rust
+#[wasm_bindgen(main)]
+fn main() -> Result<(), JsValue> {
+    Err(JsValue::from("this error message will be thrown"))
+}
+```
+
+The attribute also allows using `async fn main()` in Cargo binaries.
+
+```rust
+#[wasm_bindgen(main)]
+async fn main() {
+    // ...
+    future.await;
+}
+```
+
+This attribute is intended to be used on the `main` function of binaries or
+examples only. Unlike `#[wasm_bindgen(start)]`, it will not cause a function to
```

Comment:
```suggestion
This attribute is only intended to be used on the `main` function of binaries or
examples. Unlike `#[wasm_bindgen(start)]`, it will not cause an arbitrary function to
```

The new wording is good, just a little tweak 👍

Reply:
Done.